[AIRFLOW-6834] Fix some flaky tests in test_scheduler_job.py

yuqian90 opened the pull request:
- Fix `test_dag_file_processor_process_task_instances_depends_on_past` that sometimes fails because `TaskInstance` are returned in non-deterministic order:
```
test_scheduler_job.py::TestDagFileProcessor::test_dag_file_processor_process_task_instances_depends_on_past
```
- Fix test_task_commands.py `test_run_ignores_all_dependencies` so that it restores orm setting when done to avoid interfering with other tests.

---
Issue link: [AIRFLOW-6834](https://issues.apache.org/jira/browse/AIRFLOW-6834)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

yuqian90 committed:
[AIRFLOW-6834] Fix flaky test_scheduler_job by sorting TaskInstance

kaxil commented on airflow/models/dagrun.py:
```diff
@@ -201,7 +201,7 @@ def get_task_instances(self, state=None, session=None):
         if self.dag and self.dag.partial:
             tis = tis.filter(TaskInstance.task_id.in_(self.dag.task_ids))
-        return tis.all()
+        return tis.order_by(TaskInstance.execution_date, TaskInstance.dag_id, TaskInstance.task_id).all()
```
How about changing the test itself so order doesn't matter?

yuqian90 replied:
Good point. That's done. I changed the test instead.

yuqian90 committed:
Revert "[AIRFLOW-6834] Fix flaky test_scheduler_job by sorting TaskInstance"

This reverts commit 037dbd0e8961e404683ce03203dd0acafbb4881f.

yuqian90 committed:
Change test so order doesn't matter

yuqian90 committed:
Ensure configure_orm() is restored after test_task_command

task_run() calls configure_orm with disable_connection_pool=True.
Call configure_orm again to restore orm configuration to avoid interfering with other tests.

yuqian90 commented on airflow/cli/commands/task_command.py:
```diff
@@ -158,33 +158,32 @@ def task_run(args, dag=None):
     # behind multiple open sleeping connections while heartbeating, which could
     # easily exceed the database connection limit when
     # processing hundreds of simultaneous tasks.
-    settings.configure_orm(disable_connection_pool=True)
```
This change is needed because I realized test_task_commands sometimes interferes with other tests because this line sets `disable_connection_pool` to True without setting it back. So when test_task_commands run first, connection pool is disabled globally and never restored.
